Manual: fix uses of deprecated functions

Octachron opened the pull request:
This patch fixes the use of deprecated functions within the examples of the manual. There was two problematic subsections: one in the core language chapter ( see the first example [here](http://caml.inria.fr/pub/docs/manual-ocaml/coreexamples.html#sec12) ) due to `Array.create` and the other in the advanced examples chapter ( see the first example [here](http://caml.inria.fr/pub/docs/manual-ocaml/advexamples.html#sec52)) due to the recent separation between `bytes` and `string` .

I think it might be worthwhile to try to have a much clearer separation between `string` and `bytes` in the manual in the future. As a small step in this direction, I have added a very brief mention of the existence of `bytes` in the core language chapter and an immutable qualifier to the description of `strings`.

Octachron committed:
manual: Fix deprecated use of Array.create

Octachron committed:
manual: distinguish bytes array and string

This small fix correct the use of the deprecated String.copy in the
manual. To be more coherent, it also mention briefly the existence of
bytes and the immutability of strings in the core language section.

gasche commented:
This is a good change (I'm happy to see the merge of the manual put to good use) and I would be ready to merge it. It seems to me, however, that you haven't quite gone far enough on the `ostring` change: with immutable strings it makes little sense to have a `copy` method. Would you be motivated to rewrite this part of the manual using a `reverse` method instead of `copy`?

gasche commented on manual/manual/tutorials/coreexamples.etex:
```diff
@@ -54,20 +54,20 @@ fib 10;;
-(pronounce ``nil'') by adding elements in front using the "::"
-(``cons'') operator.
+Predefined data structures include tuples, arrays, specialized byte
+arrays and lists. General mechanisms for defining your own data
```
I would remove the "specialized byte arrays" here. They feel out of place, and indeed they are too specialized for a tutorial. The formulation (Predefined data structures _include_ ...) does not suggest that it is an exhaustive enumeration.

Octachron replied:
I agree. Fixed.

Octachron commented:
I agree that the copy method is not optimal. However, I would rather replace it with an `escaped` method or one of the others `string->string` functions that are defined in the `String` module to keep the definition of the method lightweight.

Octachron committed:
manual: revert "specialized byte array" addition

gasche commented:
I'm fine with `escaped`.

Octachron committed:
manual: improve the string class example

Replace the `copy` method, which is quite pointless for immutable
strings, with an `escaped` method.

Octachron commented:
Ok, done.

gasche commented:
Thanks!
